### PR TITLE
emacs: update patches and fix autoconf missing build failure

### PR DESCRIPTION
**Pull request by wyuenho**
Closes: https://trac.macports.org/ticket/56586
Closes: https://trac.macports.org/ticket/56564

#### Description

The emacs-app port currently is applying two patches, both have their problems:

* patch*-src_emacs.c.diff - this patch adds MacPort's prefix to the PATH of emacs but not /usr/local/bin or any other paths in the shell's environment, this patch is a half done solution to solve the PATH problem when launching Emacs.app from the Finder. A good solution to this problem (​https://github.com/purcell/exec-path-from-shell) has been around for 6 years now. I suggest we remove this, so there's less maintenance.
~* patch-enable-multicolor-fonts.diff - this patch was originally added to reenable emoji inputs, but it hasn't worked for months now. Every time an emoji is inserted, emacs crashes. I suggest we remove this patch until the emacs devs choose to reenable it.~
* update inline patch to https://gist.github.com/takaxp/5294b6c52782d0be0b25342be62e4a77
* require autoconf, automake and libtool

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.13
Xcode 9.4

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL?

- [x] checked your Portfile with `port lint`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

**Comment by l2dy** [comment]
`patch-enable-multicolor-fonts.diff` works for me (😂). Maybe you didn't choose the multicolor_font variant?
screen shot 2018-06-03

**Comment by l2dy** [comment]
You need to set the emoji font to use them though, see https://github.com/syl20bnr/spacemacs/blob/v0.200.13/layers/+os/osx/config.el#L19-L23.

**Comment by wyuenho** [comment]
Actually, the emojis do display, but inputing them crashes emacs-app. In any case, even with this multi color fonts patch and the elisp snippets, emoji support is still incomplete, this patch gives people the impression that it'll Just Work.

Users interested in emoji support should either use the emacs-map-app port or install [emojify](https://github.com/iqbalansari/emacs-emojify)

**Comment by l2dy** [comment]
I can copy & paste emojis and have never experienced a crash... This variant is not enabled by default, so I don't think it can do much harm.

File: editors/emacs/files/site-start.el [hunk 1]
```diff
@@ -10,3 +10,9 @@
 
 ;; Info-directory-list contains ${prefix}/share/info. See #32148.
 (setq Info-default-directory-list (cons "__PREFIX__/share/info" Info-default-directory-list))
+
+;; Use the OS X Emoji font for Emoticons
+(when (fboundp 'set-fontset-font)
+  (set-fontset-font "fontset-default"
+                    '(#x1F600 . #x1F64F)
```
Comment (l2dy) [inline editors/emacs/files/site-start.el]: Doing this without choosing the `multicolor_font` variant may cause Emacs to crash.

`'symbol` may be better here, some emojis were not correctly displayed for me when using `'(#x1F600 . #x1F64F)`.

**Comment by wyuenho** [comment]
Thanks deity CI's finally passed.

**Comment by pmetzger** [comment]
No response so far from @drkp. I suppose we should just merge this if the 72 hours elapse.

**Comment by drkp** [comment]
This looks good to me. I've been waiting to see if a consensus emerges on the emoji patch (which I don't know anything about since it was added before I started maintaining the -app port), but it seems like the answer is to leave it.

Thanks for fixing the autoconf issues!

**Comment by wyuenho** [comment]
Ahhh too soon

Turns out CI doesn't test subports and the patching phase failed...

I'll make a new PR

File: editors/emacs/files/patch-inline-25.2-20170426.diff [hunk 2]
```diff
@@ -802,40 +745,59 @@ diff -Nrc lisp/term/ns-win.el lisp/term/ns-win.el
...
 ***************
-*** 3875,3882 ****
+*** 2472,2481 ****
+  	swallow_events (false);		/* May clear input_pending.  */
+  
+        /* Redisplay if no pending input.  */
+!       while (!(input_pending
+! 	       && (input_was_pending || !redisplay_dont_pause)))
+  	{
+- 	  input_was_pending = input_pending;
```
Comment (l2dy) [inline editors/emacs/files/patch-inline-25.2-20170426.diff]: You forgot to bump the revision when you changed files installed by the port, which (accidentally) caused the CI not to test it.

Comment (wyuenho) [reply editors/emacs/files/patch-inline-25.2-20170426.diff]: Ah, good to know. Probably should document it somewhere.

**Pull request merged**